test/integration_test.go

```diff
@@ -60,6 +66,97 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+func TestTerragruntBeforeHook(t *testing.T) {
+	cleanupTerraformFolder(t, TEST_FIXTURE_HOOKS_BEFORE_ONLY_PATH)
```

Comment:
Ideally, we can call `t.Parallel()` at the top of each of these tests so they all run in parallel. Note that this only works if they all run in different folders.

Reply:
One other catch: the `file.out` that may of the hooks create may be left overs from previous test runs. So probably, the tests should copy the folder you're testing (e.g. `TEST_FIXTURE_HOOKS_BEFORE_ONLY_PATH`) to a new temp folder each test run to ensure we're not seeing an older `file.out`. There are several helper methods in in `integration_test.go` to do that copying.

Reply:
Good point - will look into it and implement